### ISSUE TITLE
oc_id: Rails existing process detection fails and causes high CPU utlilization.

### ISSUE THREAD

**Issue by kkeane** [chef/chef-server#403]
In Chef-server-coref 12.1.0, if the file /opt/opscode/embedded/service/oc_id/tmp/pids/server.pid exists but the corresponding process does not, rails will be caught in a restart loop, causing a high CPU utilization.

This situation happened to me if the underlying OS crashes and cannot clean up the old pid file.

**Comment by sean-horn** [comment]
Looks like this in the /var/log/opscode/oc_id/current file.
Delete the indicated file, everything goes back to normal with no restart of the oc_id service necessary

```
2015-08-11_23:53:50.42712 => Booting Unicorn
2015-08-11_23:53:50.42716 => Rails 4.2.0 application starting in production on http://localhost:9090
2015-08-11_23:53:50.42717 => Run `rails server -h` for more startup options
2015-08-11_23:53:50.42722 => Ctrl-C to shutdown server
2015-08-11_23:53:50.42780 A server is already running. Check /opt/opscode/embedded/service/oc_id/tmp/pids/server.pid.
2015-08-11_23:53:50.42783 Exiting
```

**Comment by sean-horn** [comment]
@chef/chef-visual-interaction

**Comment by hcooper** [comment]
+1, same problem. Unclean shutdown, 100% CPU after reboot.

`top -c` reporting lots of these being spawned:

    /opt/opscode/embedded/bin/ruby bin/rails server -p 9090 -b 127.0.0.1

Same entries in the log (in my case in `/opt/opscode/embedded/service/oc_id/log/current`). Fix was to just delete `/opt/opscode/embedded/service/oc_id/tmp/pids/server.pid`

Here's a screenshot of cpu utilization charts to highlight how hard it was hitting the machine:

![selection_999 149](https://cloud.githubusercontent.com/assets/131580/11319768/3fa58956-9037-11e5-8acd-79db75080b0d.png)

**Comment by legal90** [comment]
@hcooper Thanks! Removing this PID file actually helps to fix that issue.

**Comment by tatref** [comment]
Same here with version ```chef-server-core-12.5.0-1.el6.x86_64```

chef-client shows
    [2016-06-02T17:26:25+02:00] ERROR: Server returned error 502 for https://chef-server.digora.com/organizations/digora/nodes/chef-client-oel6, retrying 3/5 in 13s

I have the exact same logs under /var/log/opscode/oc_id/current

If I do ```chef-server-ctl test```

```Configuring logging...
Creating platform...
Configured URL: https://chef-server.org.com
Creating org pedant_testorg_chef-server_9851
/opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:336:in `block in create_org': Bad error code 500 from create org: {"error":["internal service error"]} (RuntimeError)
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:321:in `times'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:321:in `create_org'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:123:in `org_from_config'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:42:in `initialize'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:79:in `new'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:79:in `create_platform'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:58:in `setup'
	from ./bin/oc-chef-pedant:10:in `<main>'```

I also experience a high CPU usage, but removing the PID file is not enough, I have to restart via ```chef-server-ctl restart```, but the issue rises again after some minutes.

Can I provide anything else to help fix this bug?

**Comment by marcparadise** [comment]
Hi tatref, thanks for posting the additional info.  ` Bad error code 500 from create org` indicates that something is going wrong in the Chef Server API service. Please open a separate issue and include tail output for: 

* `/var/log/opscode-erchef/current` 
* `/var/log/opscode-erchef/crash.log.1`  (or latest, if latest is not `.1`)

**Comment by marcparadise** [comment]
Closing this as it seems to be resolved in Chef Server 12.6.0.  Performing a kill -9 on the rails processes causes runit to restart the service as expected. `oc_id/current` logs show no problem with the lingering pid file.  

I repeated the process while performing a `tail -F` on the pid file.  I was able to see that the the pid file lingered until after runit restarted oc_id, then was replaced with new content. 

Finally, I killed the process with `chef-server-ctl stop`, then created a new pid file containing an invalid process id.  `oc_id` was able to start on `chef-server-ctl start oc_id` without complications, and the pid file was updated.
